crates/conversions/src/serde/deserialize/deserialize_impl.rs

```diff
@@ -71,6 +71,27 @@ impl CairoDeserialize for bool {
     }
 }
 
+impl CairoDeserialize for NonZeroFelt {
+    fn deserialize(reader: &mut BufferReader<'_>) -> BufferReadResult<Self> {
+        let felt = reader.read::<Felt>()?;
+        NonZeroFelt::try_from(felt).map_err(|_| BufferReadError::ParseFailed)
+    }
+}
+
+impl CairoDeserialize for NonZeroU64 {
```

Comment:
Maybe let's do this with macro for convenience:

```rs
macro_rules! impl_deserialize_for_nonzero_num_type {
    ($type:ty) => {
        impl CairoDeserialize for NonZero<$type> {
            fn deserialize(reader: &mut BufferReader<'_>) -> BufferReadResult<Self> {
                let val = <$type>::deserialize(reader)?;
                NonZero::new(val).ok_or(BufferReadError::ParseFailed)
            }
        }
    };
}
```

Reply:
No need as there are only 2 impls but please do it for consistency, there are already macro calls for regular num types.